src/query/sql/src/planner/optimizer/optimizer.rs

```diff
@@ -502,115 +469,35 @@ async fn optimize_merge_into(opt_ctx: OptimizerContext, plan: Box<MergeInto>) ->
         })
     }
 
-    // try to optimize distributed join, only if
-    // - distributed optimization is enabled
-    // - no local table scan
-    // - distributed merge-into is enabled
-    if opt_ctx.enable_distributed_optimization
-        && !contains_local_table_scan(&join_sexpr, &opt_ctx.metadata)
-        && opt_ctx
-            .table_ctx
-            .get_settings()
-            .get_enable_distributed_merge_into()?
-    {
-        // distributed execution stargeties:
-        // I. change join order is true, we use the `optimize_distributed_query`'s result.
-        // II. change join order is false and match_pattern and not enable spill and not non-equal-join, we use right outer join with rownumber distributed strategies.
-        // III otherwise, use `merge_into_join_sexpr` as standalone execution(so if change join order is false,but doesn't match_pattern, we don't support distributed,in fact. case I
-        // can take this at most time, if that's a hash shuffle, the I can take it. We think source is always very small).
-        // input is a Join_SExpr
-        let mut merge_into_join_sexpr =
-            optimize_distributed_query(opt_ctx.table_ctx.clone(), &join_sexpr)?;
+    if opt_ctx.enable_distributed_optimization {
         let merge_source_optimizer = MergeSourceOptimizer::create();
-        // II.
-        // - join spilling is disabled
-        let (optimized_distributed_merge_into_join_sexpr, distributed) = if opt_ctx
```

Comment:
Is distributed supported in case of hash spill ?

Reply:
use hash join spill